fuel/src/main/kotlin/com/github/kittinunf/fuel/core/requests/UploadRequest.kt

```diff
@@ -189,3 +183,91 @@ private fun OutputStream.writeBytes(bytes: ByteArray): Long {
     write(bytes)
     return bytes.size.toLong()
 }
+
+class UploadRequest private constructor(private val wrapped: Request) : Request by wrapped {
```

Comment:
[Nits] Is there any reason we have this class in this File? Shall we extract it out like other Request counterparts?

Reply:
I would vote to split it out just for the sake of clarity and consistency to its peers. But I am not gonna say no if you have a strong reason to leave it in here.

Reply:
No strong 💪 reason, so will split it out ✂️

Reply:
Done